qiskit_ibm_runtime/runtime_job.py

```diff
@@ -453,14 +453,17 @@ def _error_msg_from_job_response(self, response: Dict) -> str:
             Error message.
         """
         status = response["state"]["status"].upper()
+        if self._status == JobStatus.ERROR:
+            error_message = self._reason if self._reason else self._error_message
         job_result_raw = self._download_external_result(
             self._api_client.job_results(job_id=self.job_id())
         )
         index = job_result_raw.rfind("Traceback")
         if index != -1:
             job_result_raw = job_result_raw[index:]
-
         error_msg = API_TO_JOB_ERROR_MESSAGE["FAILED"]
+        if not error_message is None:
+            error_msg += error_message + "\n"
         if status == "CANCELLED" and self._reason == "RAN TOO LONG":
             error_msg = API_TO_JOB_ERROR_MESSAGE["CANCELLED - RAN TOO LONG"]
         return error_msg.format(self.job_id(), job_result_raw)
```

Comment:
In the original issue, we only wanted to add the reason or error message if the result isn't returned (`job_result_raw` is none) - otherwise, we'd just be appending the `error_message` to `error_msg` which would already contain the same, duplicated error message. 

I think something like this would work: 
```suggestion
        return error_msg.format(self.job_id(), job_result_raw or self._reason)
```

Reply:
You are right. But I am thinking it may be better to leave `_error_msg_from_job_response` and instead change line 218 in `results()` as follows:
Instead of 
`error_message = self._reason if self._reason else self._error_message`
write
`error_message = self.error_message()`
What do you think?

Reply:
Yeah that would work, but I think `_error_msg_from_job_response` would still have to be updated to handle the case if `job_result_raw` is none

Reply:
I finally took your advice, with a little extra change. Can you have another look?